### PR TITLE
Sync Network Data

### PR DESCRIPTION
**Pull request by enejb**
There are some network settings that would are useful for us to know in
order to prevent some common user errors and provide a better UI when
managing sites that are part of a network.

Data gets synced every time network settings get updated and daily. 
In order for the network settings to get synced on update the user has to to have the main network site jetpack connected or have Jetpack network activated. 

Data that we are syncing.
network name, network_allow_new_registrations, add_new_users (bool), site_upload_space, upload_filetypes, fileupload_maxk and menu_items.

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -445,14 +448,25 @@ private function __construct() {
 		if( is_multisite() ) {
 			Jetpack_Network::init();
 
-			if( is_network_admin() )
-			    return; // End here to prevent single site actions from firing
+			// Only sync this info if we are on a multi site
+			// @since  3.7
+			$this->sync->mock_option( 'network_name', array( 'Jetpack', 'network_name' ) );
+			$this->sync->mock_option( 'network_allow_new_registrations', array( 'Jetpack', 'network_allow_new_registrations' ) );
+			$this->sync->mock_option( 'network_add_new_useres', array( 'Jetpack', 'network_add_new_useres' ) );
```
Comment (kraftbj) [inline class.jetpack.php]: Should this be `network_add_new_users` ?

File: class.jetpack.php [hunk 2]
```diff
@@ -885,6 +899,61 @@ function require_lib_dir( $lib_dir ) {
...
+	 * @return string
+	 */
+	static function network_allow_new_registrations( $option = null ) {
+		return ( in_array( get_site_option( 'registration' ), array('none', 'user', 'blog', 'all' ) ) ? get_site_option( 'registration') : 'none' );
+	}
+	/**
+	 * Does the network allow admins to add new users.
+	 * @return boolian
+	 */
+	static function network_add_new_useres( $option = null ) {
```
Comment (kraftbj) [inline class.jetpack.php]: Here too.

File: class.jetpack.php [hunk 3]
```diff
@@ -971,6 +1040,20 @@ function update_jetpack_main_network_site_option() {
...
 	}
+	/**
+	 * Triggered after a user updates the network settings via Network Settings Admin Page
+	 *
+	 */
+	function update_jetpack_network_settings() {
+		// Only sync this info for the main network site.
+		do_action( 'add_option_jetpack_network_name', 'jetpack_network_name', Jetpack::network_name() );
+		do_action( 'add_option_jetpack_network_allow_new_registrations', 'jetpack_network_allow_new_registrations', Jetpack::network_allow_new_registrations() );
+		do_action( 'add_option_jetpack_network_add_new_useres', 'jetpack_network_add_new_useres', Jetpack::network_add_new_useres() );
```
Comment (kraftbj) [inline class.jetpack.php]: Last instance :)

Comment (samhotchkiss) [inline class.jetpack.php]: At least he was consistent :)

Comment (enejb) [inline class.jetpack.php]: in my language it makes sense :)

**Comment by enejb** [comment]
Thanks for the review @kraftbj I made the changes to the PR.

**Comment by zinigor** [comment]
Read the changes, tested the sync, it runs without any problems.

**Pull request merged**